behat/tests/bootstrap/v2/reporting/sections/MoneyOutShortSectionTrait.php

```diff
@@ -0,0 +1,332 @@
+
+        if (!$categoryRows) {
+            $this->throwContextualException('A div element was not found on the page');
+        }
+
+        if (count($this->categoryList) < 1) {
+            $this->assertStringContainsString('None', $categoryRows[1]->getHtml(), 'Short Money Out Categories');
+        } else {
+            $categoryListItems = $categoryRows[1]->findAll('css', 'li');
+            foreach ($this->categoryList as $expectedCategoryKey => $expectedCategory) {
+                $this->assertStringContainsString($expectedCategory, $categoryListItems[$expectedCategoryKey]->getHtml(), 'Short Money Out Categories');
+            }
+        }
+    }
+
+    private function checkOneOffPaymentsExist($oneOffPayments)
+    {
+        $oneOffPaymentYesNo = $oneOffPayments->findAll('css', 'div.govuk-summary-list__row');
+
+        if (!$oneOffPaymentYesNo) {
+            $this->throwContextualException('A div element was not found on the page');
+        }
+
+        if (count($this->oneOffPaymentsList) < 1) {
+            $this->assertStringContainsString('No', $oneOffPaymentYesNo[1]->getHtml(), 'Short Money Out One Off Payments');
+        } else {
+            $this->assertStringContainsString('Yes', $oneOffPaymentYesNo[1]->getHtml(), 'Short Money Out One Off Payments');
+        }
+    }
+
+    private function checkOneOffPaymentRows($oneOffPaymentTBodyRows)
+    {
+        $oneOffPaymentRows = $oneOffPaymentTBodyRows[0]->findAll('css', 'tr');
+        $oneOffPaymentTotalRow = $oneOffPaymentTBodyRows[1];
+
+        foreach ($oneOffPaymentRows as $oneOffPaymentRowKey => $oneOffPaymentRow) {
+            $dateTimestamp = sprintf(
+                '%s-%s-%s 00:00',
+                $this->oneOffPaymentsList[$oneOffPaymentRowKey]['year'],
+                $this->oneOffPaymentsList[$oneOffPaymentRowKey]['month'],
+                $this->oneOffPaymentsList[$oneOffPaymentRowKey]['day']
+            );
+
+            $date = date('j F Y', strtotime($dateTimestamp));
+            $amountFormatted = $this->moneyFormat($this->oneOffPaymentsList[$oneOffPaymentRowKey]['amount']);
+
+            $this->assertStringContainsString(
+                $this->oneOffPaymentsList[$oneOffPaymentRowKey]['description'],
+                $oneOffPaymentRow->getHtml(),
+                'Short Money Out One Off Payments Items'
+            );
+            $this->assertStringContainsString($amountFormatted, $oneOffPaymentRow->getHtml(), 'Short Money Out One Off Payments Items');
+            $this->assertStringContainsString($date, $oneOffPaymentRow->getHtml(), 'Short Money Out One Off Payments Items');
+
+            $this->oneOffPaymentsTotal += floatval($this->oneOffPaymentsList[$oneOffPaymentRowKey]['amount']);
+        }
+
+        $this->assertStringContainsString($this->moneyFormat($this->oneOffPaymentsTotal), $oneOffPaymentTotalRow->getHtml(), 'Short Money Out One Off Payments Total');
```

Comment:
May have missed it but are we asserting on the description of the payment here?

Reply:
yeah it just got formatted a bit funny cos it's a long line: $this->oneOffPaymentsList[$oneOffPaymentRowKey]['description']